athina/evals/base_evaluator.py

```diff
@@ -0,0 +1,189 @@
+from abc import ABC, abstractmethod
+from concurrent.futures import ThreadPoolExecutor, as_completed
+from typing import Any, Dict, List
+from athina.helpers import logger
+from athina.helpers.athina_logging_helper import AthinaLoggingHelper
+from athina.helpers.type_check_helper import TypeCheckHelper
+from athina.interfaces.athina import AthinaExperiment
+
+from athina.interfaces.data import DataPoint
+from athina.interfaces.result import BatchRunResult
+from athina.services.athina_api_service import AthinaApiService
+
+
+class BaseEvaluator(ABC):
+    def __init__(self):
+        pass    
+
+    # Abstract properties
+    @property
+    @abstractmethod
+    def name(self) -> str:
+        """A unique name identifier for the evaluator."""
+        pass
+
+    @property
+    @abstractmethod
+    def display_name(self) -> str:
+        """A display name for the evaluator."""
+        pass
+
+    @property
+    @abstractmethod
+    def metric_ids(self) -> str:
+        """The metric computed by the evaluator."""
+        pass
+
+    @property
+    @abstractmethod
+    def required_args(self):
+        """A list of required arguments for the evaluator."""
+        pass
+
+    @property
+    @abstractmethod
+    def default_model(self):
+        """The default model for the evaluator."""
+        pass
```

Comment:
We should move this into the `LlmEvaluator` class instead of having it here.

Reply:
Done